src/Nethermind/Nethermind.Evm/VirtualMachine.cs

```diff
@@ -2410,7 +2404,7 @@ private EvmExceptionType InstructionSelfDestruct<TTracing>(EvmState vmState, ref
 
         bool accountExists = _state.AccountExists(contractAddress);
         if (accountExists && (_codeInfoRepository.GetCachedCodeInfo(_worldState, contractAddress, spec).MachineCode.Length != 0 ||
-                              _state.GetNonce(contractAddress) != 0))
+                              _state.GetNonce(contractAddress) != 0 || _worldState.GetStorageRoot(contractAddress) != Keccak.EmptyTreeHash))
```

Comment:
looks like it is the same conditions here and in `PrepareAccountForContractDeployment`, can you make it into a method?

Reply:
Yep